src/microphysics/opac_phoebus/opac_phoebus.hpp

```diff
@@ -19,8 +19,110 @@
 
 using namespace parthenon::package::prelude;
 
+// singularity includes
+#include <singularity-opac/base/radiation_types.hpp>
+#include <singularity-opac/neutrinos/mean_opacity_neutrinos.hpp>
+#include <singularity-opac/neutrinos/mean_s_opacity_neutrinos.hpp>
+#include <singularity-opac/neutrinos/opac_neutrinos.hpp>
+#include <singularity-opac/neutrinos/s_opac_neutrinos.hpp>
+
 namespace Microphysics {
 
+using RadiationType = singularity::RadiationType;
+
+/// One class to contain all opacity objects and wrap the subset of calls we use in
+/// phoebus for convenience.
+class Opacities {
+  using Opacity = singularity::neutrinos::Opacity;
+  using MeanOpacity = singularity::neutrinos::MeanOpacity;
+  using SOpacity = singularity::neutrinos::SOpacity;
+  using MeanSOpacity = singularity::neutrinos::MeanSOpacity;
+
+ public:
+  Opacities(const Opacity &opac, const MeanOpacity &m_opac, const SOpacity &s_opac,
+            const MeanSOpacity &m_s_opac)
+      : opac_(opac), m_opac_(m_opac), s_opac_(s_opac), m_s_opac_(m_s_opac) {}
```

Comment:
@Yurlungur this is actually failing on GPUs for me, I'm pretty sure at the first place any opacity functions are called, when calling `Opacities::EnergyDensityFromTemperature` in the torus pgen file, I get a
`return index_ == static_cast<index_t<Ts...>>(-1);` from `variant`. Do you think I need to do anything else to make `Opacities` available on device, like the singularity classes `GetOnDevice {return *this;}` functions?

Reply:
Wait are you constructing this `Opacities` object on device? Or constructing it on host and then passing it around? In either case, I can see two possible issues:
1. The default copy constructor is deleted by the compiler if you have const fields. Make your fields non-const (not as args to the constructor, down where you declare them at the bottom of the struct.)
2. Similarly, if you are paccing the `Opacities` struct into kernels because you constructed it on host, the fields can't be references because they need to be captured by value by the lambda.